javascript/javascript3/week3/ShoppingCart/shoppingcart-.js

```diff
@@ -0,0 +1,140 @@
+const body = document.getElementsByTagName("body");
+const h1 = document.getElementById("whos-shopping-cart");
+const itemList = document.getElementById("item-list");
+// const itemAddBtn = document.getElementById("add-btn");
+const cartTotal = document.getElementById("cart-total");
+const searchInput = document.getElementById("product-search-input");
+
+let cartTotalPrice = 0;
+
+class Product {
+  constructor(name, price) {
+    this.name = name;
+    this.price = price;
+  }
+
+  convertToCurrency(currency) {
+    // Assuming dkr as default currency
+    try {
+      fetch(
+        `https://v6.exchangerate-api.com/v6/a534b80e56a63f91e332441c/latest/DKK`
+      )
+        .then((response) => response.json())
+        .then((data) => {
+          console.log(data);
+          const allCurrencies = data.conversion_rates[currency];
+          console.log(allCurrencies);
+          const changeMoney = this.price * allCurrencies;
+          console.log(Math.ceil(changeMoney) + " won");
+        });
+    } catch (err) {
+      console.log(err);
+    }
+  }
+}
+
+const plant = new Product("plant", 50);
+console.log(plant.convertToCurrency("KRW"));
+
+const figs = new Product("fig", 30);
+console.log(figs.convertToCurrency("KRW"));
+
+const plantMilk = new Product("soy milk", 20);
+console.log(plantMilk.convertToCurrency("KRW"));
+
+class ShoppingCart {
+  constructor() {
+    this.products = [];
+  }
+
+  addProduct(product) {
+    // should add a product to the products array.
+    this.products.push(product);
+  }
+
+  removeProduct(product) {
+    // should remove a product from the products array
+    const findIndex = this.products.indexOf(product);
+    if (findIndex > -1) {
+      this.products.splice(product, 1);
+    }
+  }
+
+  searchProduct(productName) {
+    // should return an array of product that match the productName parameter
+    const filteredItems = this.products.filter((item) =>
+      item.name.toLowerCase().includes(productName)
+    );
+    return filteredItems;
+  }
+
+  getTotal() {
+    // should get the total price of the products in the shoppingcart
+    const items = this.products;
+
+    const result = items
+      .map((itemPrice) => itemPrice.price)
+      .reduce((total, prev) => total + prev, 0);
+    return result;
+  }
+
+  renderProducts(showProduct) {
+    // should render the products to html. You decide what to show and how
+
+    showProduct.forEach((product) => {
+      const productList = document.createElement("ul");
+      const productAdd = document.createElement("button");
+      productList.innerHTML = `<li><h3>${product.name}</h3> <h4>${product.price} dkk</h4></li> `;
+      productAdd.innerHTML = `Add Product`;
+      itemList.append(productList);
+      productList.append(productAdd);
+
+      productAdd.addEventListener("click", () => {
+        cartTotalPrice += product.price;
+        cartTotal.innerHTML = cartTotalPrice;
+      });
+    });
+  }
+
+  getUser() {
```

Comment:
either this should be called `getRandomUser()` or it should take a number as a parameter `getUser(number)` and you skip the random number generation.